src/movies.js

```diff
@@ -1,16 +1,84 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(movies) {
+    const directors = movies.map(function(movie) {
+      return movie.director;
+    })
+    return directors;
+  };
+  
+  getAllDirectors(movies);
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(movies) {
+    const moviesOfSpielberg = movies.filter(function(SpielbergMovie) {
+      return SpielbergMovie.director==="Steven Spielberg" && SpielbergMovie.genre.includes("Drama");;
+    })
+    return moviesOfSpielberg.length;
+  };
+  
+  howManyMovies(movies);
+
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(movies) {
+    const averageWithAllDecimals = movies.reduce(function(sum, val) {
+      return sum+val.rate/movies.length;
+    }, 0)
```

Comment:
here you needed to check in some of the elements if they had any rate at all because some of the movies didnt have any rate so you could have created a small if statement in the reduce fuction to check that part:
```
if (typeof val.rate==="number") return sum+val.rate;
        else return sum;
```